pkg/operator/controllers/banner/banner.go

```diff
@@ -0,0 +1,43 @@
+package banner
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+
+	consolev1 "github.com/openshift/api/console/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+)
+
+func (r *Reconciler) reconcileBanner(ctx context.Context, instance *arov1alpha1.Cluster) (err error) {
+	switch instance.Spec.Banner.Content {
+	case arov1alpha1.BannerDisabled:
+		err = r.consolecli.ConsoleV1().ConsoleNotifications().Delete(ctx, BannerName, metav1.DeleteOptions{})
+		if err != nil && kerrors.IsNotFound(err) {
+			// we don't care if the object doesn't exist
+			err = nil
```

Comment:
It is easy to break error handling like. We definitely need a unit test.

Also I suggest considering writing an e2e test (separate PR is fine) where we enable a banner, check that the banner object exists, diable it and check that the object is no longer exists.

UPD: Clarification for unit test, I think we should test via `Reconcile` method to test whole flow, not only `reconcileBanner`.

Reply:
Added unit test for Reconcile. In this case it's almost a 1:1 test with reconcileBanner.